var/spack/repos/builtin/packages/intel-mpi/package.py

```diff
@@ -75,15 +81,40 @@ def setup_dependent_environment(self, spack_env, run_env, dependent_spec):
         spack_env.set('I_MPI_FC', spack_fc)
 
     def setup_dependent_package(self, module, dep_spec):
-        # Check for presence of bin64 or bin directory
-        if os.path.isdir(self.prefix.bin):
-            bindir = self.prefix.bin
-        elif os.path.isdir(self.prefix.bin64):
-            bindir = self.prefix.bin64
+        bindir = self.prefix.compilers_and_libraries.linux.mpi.intel64.bin
+
+        if self.compiler.name == 'intel':
+            self.spec.mpicc  = bindir.mpiicc
+            self.spec.mpicxx = bindir.mpiicpc
+            self.spec.mpifc  = bindir.mpiifort
+            self.spec.mpif77 = bindir.mpiifort
```

Comment:
Fixes #4630. Until someone can come up with a better solution we should just use `mpiicc` and friends. That's what EasyBuild does.